### PR TITLE
specs, fix azure example route

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/azure/example/basic/client.tsp [hunk 1]
```diff
@@ -2,9 +2,11 @@ import "@azure-tools/typespec-client-generator-core";
 
 import "./main.tsp";
 
+using TypeSpec.Http;
 using Azure.ClientGenerator.Core;
 
 @TypeSpec.Versioning.useDependency(_Specs_.Azure.Example.Basic.Versions.v2022_12_01_preview)
+@route("/azure/example/basic")
 namespace Client;
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/example/basic/client.tsp:10]: `@scenarioService` add this route to namespace in main.tsp
https://github.com/azure/cadl-ranch/blob/main/packages/cadl-ranch-expect/src/decorators.ts#L242

Therefore, I need to add the same `@route` to this namespace in client.tsp as well.

@timotheeguerin Would you let me know if this is the correct understanding and the correct fix to make the route same (via main.tsp and via client.tsp)?

Comment (timotheeguerin) [reply packages/cadl-ranch-specs/http/azure/example/basic/client.tsp:10]: yeah that would be the way but is it necessary to have a client.tsp here or can't you do `@@client` on the service namespace. Leave client structure testing to another test

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/example/basic/client.tsp:10]: Got it. Thanks Timothee.

The client.tsp is here intended to verify the correctness of `operationId` (as interface/operation in client.tsp can provide a false `operationId: AzureExampleClient_BasicAction`. The intention is that example loader find the correct one in main.tsp as `operationId: ServiceOperationGroup_Basic`.

**Review by qiaozha — APPROVED** [review]

File: packages/cadl-ranch-specs/http/azure/example/basic/client.tsp (hunk 1 above)
Comment (qiaozha) [reply packages/cadl-ranch-specs/http/azure/example/basic/client.tsp:10]: I thought in this case `_Specs_.Azure.Example.Basic`, but it was not, which is contradict with my understanding of global namespace.

**Review by tadelesh — APPROVED** [review]

File: packages/cadl-ranch-specs/http/azure/example/basic/client.tsp (hunk 1 above)
Comment (timotheeguerin) [reply packages/cadl-ranch-specs/http/azure/example/basic/client.tsp:10]: @qiaozha no sure i get the problem here can you share more details?

Comment (qiaozha) [reply packages/cadl-ranch-specs/http/azure/example/basic/client.tsp:10]: Here we have specified the endpoint, which should be a host template right ? https://github.com/Azure/cadl-ranch/pull/655/files#diff-ddd397f87276d566b5395a2c63595c897126244510eede7bc2675652bb616af1L13 
and operation route path should be `/azure/example/basic/basic` and the final url should be endpoint host template + operation route path, we get the `/azure/example/basic/azure/example/basic/basic`.
However, we got `/azure/example/basic/azure/example/basic/basic` directly in the operation route path and the host template is empty if we read from main.tsp, but if we read from client.tsp, the operation route path is  `/azure/example/basic/basic`, and still no host template.

Comment (timotheeguerin) [reply packages/cadl-ranch-specs/http/azure/example/basic/client.tsp:10]: `@serviceScenario` just calls `@route` with that value. When you do `op is` it doesn't pull in values from containers so this change just copies over the info that is not carried over

**Pull request merged**